tiled/adapters/arrow.py

```diff
@@ -0,0 +1,322 @@
+        ]
+        return assets
+
+    def structure(self) -> TableStructure:
+        """
+
+        Returns
+        -------
+
+        """
+        return self._structure
+
+    def get(self, key: str) -> Union[ArrayAdapter, None]:
+        """
+
+        Parameters
+        ----------
+        key :
+
+        Returns
+        -------
+
+        """
+        if key not in self.structure().columns:
+            return None
+        return ArrayAdapter.from_array(self.read([key])[key].values)
+
+    def generate_data_sources(
+        self,
+        mimetype: str,
+        dict_or_none: Callable[[TableStructure], Dict[str, str]],
+        item: Union[str, Path],
+        is_directory: bool,
+    ) -> List[DataSource]:
+        """
+
+        Parameters
+        ----------
+        mimetype :
+        dict_or_none :
+        item :
+        is_directory :
+
+        Returns
+        -------
+
+        """
+        return [
+            DataSource(
+                structure_family=self.structure_family,
+                mimetype=mimetype,
+                structure=dict_or_none(self.structure()),
+                parameters={},
+                management=Management.external,
+                assets=[
+                    Asset(
+                        data_uri=ensure_uri(item),
+                        is_directory=is_directory,
+                        parameter="data_uris",  # <-- PLURAL!
+                        num=0,  # <-- denoting that the Adapter expects a list, and this is the first element
+                    )
+                ],
+            )
+        ]
+
+    #
+    @classmethod
+    def from_single_file(
+        cls,
+        data_uri: str,
+        structure: Optional[TableStructure] = None,
+        metadata: Optional[JSON] = None,
+        specs: Optional[List[Spec]] = None,
+        access_policy: Optional[AccessPolicy] = None,
+    ) -> "ArrowAdapter":
+        """
+
+        Parameters
+        ----------
+        data_uri :
+        structure :
+        metadata :
+        specs :
+        access_policy :
+
+        Returns
+        -------
+
+        """
+        return cls(
+            [data_uri],
+            structure=structure,
+            metadata=metadata,
+            specs=specs,
+            access_policy=access_policy,
+        )
+
+    def __getitem__(self, key: str) -> ArrayAdapter:
+        """
+
+        Parameters
+        ----------
+        key :
+
+        Returns
+        -------
+
+        """
+        # Must compute to determine shape.
+        return ArrayAdapter.from_array(self.read([key])[key].values)
+
+    def items(self) -> Iterator[Tuple[str, ArrayAdapter]]:
+        """
+
+        Returns
+        -------
+
+        """
+        yield from (
+            (key, ArrayAdapter.from_array(self.read([key])[key].values))
+            for key in self._structure.columns
+        )
+
+    @property
+    def reader_handle(self) -> List[pyarrow.RecordBatchFileReader]:
```

Comment:
I think this is not actually helpful and that the file-opening code should be moved into `read` and `read_partition` directly. I'll comment more on why below.

I believe this idea came from imitating the approach used in CSV. In the case of CSV we were leveraging the "laziness" (deferred execution) of `dask.dataframe.read_csv` and that does not apply here. We can do something simpler. :-)